source/common/config/subscription_factory_impl.cc

```diff
@@ -86,8 +86,12 @@ SubscriptionPtr SubscriptionFactoryImpl::subscriptionFromConfigSource(
     }
   }
   case envoy::config::core::v3::ConfigSource::ConfigSourceSpecifierCase::kAds: {
+    Config::GrpcMuxSharedPtr gmuxsp = cm_.adsMux();
+    if ( gmuxsp == nullptr ) {
+      throw EnvoyException("A configuration for dynamic resources is required for ads with a non primary cluster");
```

Comment:
Hmm, wasn't the issue here that sub-components (like SDS) of a primary cluster cannot be configured via ADS?

Reply:
Agree with @dmitri-d 
We could fix SDS, or explicitly disable SDS with primary cluster for now **as a workaround**.

Reply:
We'd have to make SDS aware of the context in which it's being configured, which is a bit ugly; we could miss this issue in future components too if we handled this misconfiguration in SDS.

I'm not sure if it makes sense to continue launching Envoy if we hit this issue, baling with a very clear explanation about what the problem is looks like a good approach to me.